### PR TITLE
Translating ch06-02 in French.

### LINKED ISSUE #59: Translating ch06-02
I'm working on the French translation of this page.

### PR DESCRIPTION
**Pull request by Jimskapt**
Closes #59

### CODE REVIEW

File: FRENCH/src/ch06-02-match.md [hunk 1]
```diff
@@ -0,0 +1,673 @@
...
+valeurs possibles que nous ne traitons pas avant le caractère réservé `_`.
+
+<!--
+However, the `match` expression can be a bit wordy in a situation in which we
+care about only *one* of the cases. For this situation, Rust provides `if let`.
+-->
+
+Cependant, l'expression `match` peut être un peu verbeuse dans une situation où
+nous nous préoccupons uniquement *d'un seul* cas. Pour ce cas, Rust nous propose
+d'utiliser `if let`.
```
Comment (LeSeulArtichaut) [inline FRENCH/src/ch06-02-match.md]:
```suggestion
d'utiliser une autre structure, `if let`.
```

File: FRENCH/src/ch06-02-match.md [hunk 2]
```diff
@@ -0,0 +1,673 @@
...
+branches, le `_` va correspondre à tous les autres cas qui ne sont pas listés
+avant. Au final, nous avons pu dire que nous ne voulons rien faire pour les
+valeurs possibles que nous ne traitons pas avant le caractère réservé `_`.
+
+<!--
+However, the `match` expression can be a bit wordy in a situation in which we
+care about only *one* of the cases. For this situation, Rust provides `if let`.
+-->
+
+Cependant, l'expression `match` peut être un peu verbeuse dans une situation où
```
Comment (LeSeulArtichaut) [inline FRENCH/src/ch06-02-match.md]: Even if `verbeux` does exist in French, I'm not sure that non-english speakers can guess its meaning if they don't know the word without doing research, I don't think we want that to happen

File: FRENCH/src/ch06-02-match.md [hunk 3]
```diff
@@ -0,0 +1,673 @@
...
+
+### Les `match` sont toujours exhaustifs
+
+<!--
+There’s one other aspect of `match` we need to discuss. Consider this version
+of our `plus_one` function that has a bug and won’t compile:
+-->
+
+Il y a un autre point de `match` que nous devons aborder. Admettons que cette
+version de notre fonction `plus_un` a un bogue et ne va pas se compiler :
```
[the remark this answers is not in the document]
Comment (Jimskapt) [reply FRENCH/src/ch06-02-match.md]: We have made the choice to translate a maximum of words in french, so all the previous and next pages uses already `bogue`.
By the way, I agree with you, a hudge majority of people uses this anglicism, but I'm the one who use this word in personal et professional life.
I think for this case if we chose one or another, it will not be difficult for our readers to understand the meaning of this word :smile:
If this comes as a shock to you, I'm disposed to make this change but it involves far more work than let it like that :wink:

File: FRENCH/src/ch06-02-match.md (hunk 2 above)
Comment (Jimskapt) [reply FRENCH/src/ch06-02-match.md]: Okay, what do you think of `lourde` or ` pompeuse`, in this case ?

File: FRENCH/src/translation-terms.md [hunk 4]
```diff
@@ -78,6 +80,7 @@ français.
 | library crate | crate de bibliothèque | s'utilise au féminin |
 | lifetime | durée de vie | - |
 | linker | linker | - |
+| literal value | valeur pure | - |
```
[the remark this answers is not in the document]
Comment (Jimskapt) [reply FRENCH/src/translation-terms.md]: In previous translation, we already think that and changed it to `valeur littérale` (I've found the first real occurrence of `literal` in line 455 of ch04-01), is it ok for you ?

File: FRENCH/src/ch06-02-match.md (hunk 2 above)
Comment (LeSeulArtichaut) [reply FRENCH/src/ch06-02-match.md]: I'd say `lourde` or `inutilement longue` could fit well here

File: FRENCH/src/translation-terms.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (LeSeulArtichaut) [reply FRENCH/src/translation-terms.md]: Yes, I think it is ok

File: FRENCH/src/ch06-02-match.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (LeSeulArtichaut) [reply FRENCH/src/ch06-02-match.md]: We should decide whether to change it or not, without looking at how much of an effort it is to make the change. For the chapters already released to the `french-release` branch, we just have to take any text editor and `find and replace in project` 😄 . For the pending PRs, the change can be made during proofreading.

@joeljpresent What is your opinion about this one?

**Comment by LeSeulArtichaut** [comment]
@Jimskapt Chapters should be merged and squashed in order to avoid flooding the commit history. Also, when trying to resolve conflicts from one chapter to another, it would be better to rebase the branch instead of merging to avoid the additional merge commit.

File: FRENCH/src/ch06-02-match.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (joeljpresent) [reply FRENCH/src/ch06-02-match.md]: Whilst *bug* is more common than *bogue*, it is common in French translations of documentations to use "pure" French equivalents rather than anglicisms.

Besides, using *bug* rather than *bogue* has a drawback: how do you translate the verb *to bug*?
*Buger*, *bugger*, *buguer*, *bugguer*, *beuguer*...?
Whereas using *bogue* makes the translation straightforward: *boguer*, as well as the translation of *debugger* to *débogueur*.
In that sense, keeping *bogue* is beneficial.

[the remark this answers is not in the document]
Comment (LeSeulArtichaut) [reply FRENCH/src/ch06-02-match.md]: Right.
@Jimskapt this review can be marked resolved, so we are able to merge the PR and decide whether to replace `bogue` with `bug` in a future discussion/PR

[the remark this answers is not in the document]
Comment (Jimskapt) [reply FRENCH/src/ch06-02-match.md]: nice, I'll do changes then merging tomorrow
